src/classes/COFF_Affiliation_TDTM.cls

```diff
@@ -462,6 +474,8 @@ public class COFF_Affiliation_TDTM extends TDTM_Runnable {
     ********************************************************************************************************/
     @testVisible
     private void updateConnectionsOnFacultyRemove(List<ID> removedFacultyIDs, Set<Id> offsWithRemovedFaculty, List<SObject> objectsToUpdate) {
+        // TODO: This doesn't delete anything, just updates course connections - should it actually delete affiliations and course connections, 
```

Comment:
I found this to be confusing -- the ApexDocs above contradict what this method actually does. I think we need to either update the comments or update the logic -- probably best to create a tech debt work item, perhaps?

Reply:
I need to review this further. It looks like you had to add the connection to the list to be updated when it suggests this method was doing that already, so I can see why you were confused. I need to take a look at what was going on here before (outside of githubs UI because that's just difficult to read). 
In any case, if the ApexDocs are saying one thing and we are doing another we should certainly create a Tech Debt WI for this.

Reply:
The only code change I made here was to only add the object to objectsToUpdate if it actually needed to be updated.

I was just really confused because the ApexDocs say "Deletes the Affiliations that had been automatically created from the Course Enrollments with assigned Faculty members. Also deletes the Course Enrollments themselves." -- when it does literally none of that.

Reply:
Yea we really need to fix the apex docs. We may want to involved Ali with this and find out what it is really supposed to be doing and either fix the docs or the code.

Reply:
I'm playing a bit of git blame / history tracking to see if I can figure out when this was done. I'll reach out to him next week once he's back.

Reply:
@lifewithryan  I did some digging and this conflicting ApexDocs / functionality has been around since 2016. I did some testing and it works like this;
- This method is only called when course connections is enabled and the Primary Faculty lookup is cleared on the Course Offering. So it makes sense to update the Course Connection to no longer be primary, and to keep the data.
- When a Course Offering is deleted from the system, then the "deleteAfflsEnrolls" method (another method in this class) is called, which actually does delete Affiliations and Course Connection records.

In my opinion, I think it's safe to say that the ApexDocs are wrong and need to be updated. The functionality is correct, unless we don't want to keep historical data. But that seems to go against how EDA functions everywhere else.

Reply:
@lifewithryan If you agree with this, I can go ahead and update the ApexDocs here now. We can circle back with Ali & Dina next week.

Reply:
mighty fine detective work. If we can update the docs now, I say we do so.

Reply:
Done, resolving